src/extra/python/scripts/remove_certain_restart_and_data_files.py

```diff
@@ -12,46 +13,58 @@ def __init__(self, basedir, workdir, datadir, exp_name):
+    basedir = os.environ['GFDL_BASE']
     expname = '/'+exp_name+'/'
 
     exp_object = temporary_exp_object(basedir, workdir, datadir, exp_name)
     
     return exp_object
 
 
-def keep_only_certain_restart_files(exp_object, max_num_files, interval=12):
+# def keep_only_certain_restart_files(exp_object, max_num_files, interval=12):
 
-    #        sh.ls(sh.glob(P(self.workdir,'restarts','res_*.cpio'))) #TODO get max_num_files calculated in line, rather than a variable to pass.
+#     #        sh.ls(sh.glob(P(self.workdir,'restarts','res_*.cpio'))) #TODO get max_num_files calculated in line, rather than a variable to pass.
 
-            #First defines a list of ALL the restart file numbers
-        files_to_remove=list(range(0,max_num_files))
+#             #First defines a list of ALL the restart file numbers
+#         files_to_remove=list(range(0,max_num_files))
 
-            #Then defines a list of the ones we want to KEEP
-        files_to_keep  =list(range(0,max_num_files,interval)) 
+#             #Then defines a list of the ones we want to KEEP
+#         files_to_keep  =list(range(0,max_num_files,interval)) 
 
-            #Then we remove the files we want to keep from the list of all files, giving a list of those we wish to remove
-        for x in files_to_keep:
-               files_to_remove.remove(x) 
+#             #Then we remove the files we want to keep from the list of all files, giving a list of those we wish to remove
+#         for x in files_to_keep:
+#                files_to_remove.remove(x) 
 
-            #Then we remove them.
-        for entry in files_to_remove:
-            try:
-                sh.rm(P(exp_object.workdir,exp_object.expname,'restarts','res_'+str(entry)+'.cpio'))
-#                 print P(exp_object.workdir,exp_object.expname,'restarts','res_'+str(entry)+'.cpio')
+#             #Then we remove them.
+#         for entry in files_to_remove:
+#             try:
+#                 sh.rm(P(exp_object.workdir,exp_object.expname,'restarts','res_'+str(entry)+'.cpio'))
+# #                 print P(exp_object.workdir,exp_object.expname,'restarts','res_'+str(entry)+'.cpio')
 
-            except sh.ErrorReturnCode_1:
-                pass
-#                 print 'Tried to remove some restart files, but number '+str(entry)+' does not exist'
+#             except sh.ErrorReturnCode_1:
+#                 pass
+# #                 print 'Tried to remove some restart files, but number '+str(entry)+' does not exist'
                 
-def keep_only_certain_restart_files_data_dir(exp_object, max_num_files, interval=12):
+def keep_only_certain_restart_files_data_dir(exp_object, max_num_files=None, interval=12):
 
     #        sh.ls(sh.glob(P(self.workdir,'restarts','res_*.cpio'))) #TODO get max_num_files calculated in line, rather than a variable to pass.
 
+        if max_num_files is None:
+            month_list = glob(P(exp_object.datadir,exp_object.expname, 'restarts')+'/res*.tar.gz')
```

Comment:
On a non-unix server (like if you are running on a local machine for example) would it still create a tar.gz?